packages/mobile/src/qrcode/QRScanner.tsx

```diff
@@ -49,6 +50,19 @@ class QRScanner extends React.Component<Props> {
       return
     }
     this.setState({ camera: true })
+    this.setState({ qrSubmitted: false })
+  }
+
+  onBardCodeDetected = (rawData: any) => {
+    if (
+      rawData.type === 'barcode' &&
+      rawData.barcodes.length !== 0 &&
+      this.state.qrSubmitted === false
+    ) {
+      this.setState({ qrSubmitted: true })
+      this.props.handleBarcodeDetected(rawData.barcodes[0])
```

Comment:
I recommend moving this to `setState`'s callback or we could still get multiple calls before the state actually gets updated